### DISCUSSION TITLE
Definition of active users

### DISCUSSION THREAD

**Discussion by nemobis** [mastodon/mastodon#18920] [Q&A] [answered]
What are the "active users" in the admin dashboards and node statistics?

In my instance, I see that almost 100 % of registered users are considered "active users", even though I know for sure that some 20 % of them didn't post anything for months.

From https://github.com/mastodon/mastodon/issues/874 and ActivityTracker calls in https://github.com/mastodon/mastodon/blob/v3.5.3/app/models/user.rb it looks like a user might be registered as active any time they log in, is that right?

**Comment by ClearlyClaire** [comment, marked as answer]
Yes, “active users” tracks users that have been logged in at one point during a period of time. Please also note that suspended or deleted accounts are not removed from the active user count.

**Reply by nemobis** [reply]
That's ok (if confusing) but it would be nice to add more statistics: https://github.com/mastodon/mastodon/issues/25700

**Comment by cham2019** [comment]
Thank you for posting this question (and the replied answer).  I finally found what I was looking for.  

I had created an instance, for tests, so I used a dummy.domain first.  Created 3 accounts, (1 admin, 2 users).  The counter was showing of course "3 active users".  Then I deleted those 2 user accounts, just leaving the server admin.  Only 1 active user in the server.  The counter still showing 3 (which I guess it was right, since it was taking into the account the former existing accounts that logged into the server during the timeframe the counter is considering).  
Finally, the test server worked fine.  So I decided to erase the mastodon directory and empty the mastodon postgres database, to re-install, re-compile assets again (mastodon entire directory was re-created all over again).  This time with the production.domain.   Server is working perfectly fine, connects with fediverse accounts, etc.  However, this time I haven't created any other user account. I just started this server with just the Admin account alone in the server (again, new database from scratch).  The counter is showing "3" active users.

Then I created 1 user test account, but I deleted after that.  Today, I loaded the main page, and the counter increased to 4 users. 

I just read from the link that the OP shared above:  https://github.com/mastodon/mastodon/blob/v3.5.3/app/models/user.rb  :

_**"# The home and list feeds will be stored in Redis for this amount of time, and status fan-out to followers will include only people within this time frame."**_

So apparently the initial "3" counting (from former test.domain server) was saved somewhere in the server cache (out of the database, out of the mastodon directory) when I re-installed mastodon from scratch, as far as I understand this counter behavior. 

I just edited the "7" days to "1", in the  /home/mastodon/live/app/models/user.rb file   to see what happens by tomorrow.

**Reply by nemobis** [reply]
It's a known issue (can't find the report) that counts aren't always decremented with deletions etc.

But also, have you tried restarting *all* services (including postgresql and redis)? I noticed that some statistics on my instance were suspiciously low, but magically they returned to more plausible figures after restarting postgresql.

**Reply by cham2019** [reply]
Yeah, I have tried restarting all the services (mastodon, postgres, and redis), as well as rebooting the entire server.  Same stats number. 
I recreated the DB file from the Redis directory (by deleting it, and restarting Redis).  Same stats number.
So, now I'm waiting to see what happens tomorrow, with the number of days "1"  on the ACTIVE_DURATION line in the /home/mastodon/live/app/models/user.rb file.
